tpchgen-cli/src/main.rs

```diff
@@ -411,69 +419,6 @@ impl Cli {
         File::create(path)
     }
 
-    /// Returns a list of "parts" (data generator chunks, not TPCH parts) to create
```

Comment:
I simply moved this logic into `GenerationPlan::new()` -- I tried to keep the algorithm the same in this PR (though I expect we will change the logic in subsequent PRs)